integration/helpers/utils.ts

```diff
@@ -69,71 +34,45 @@ export function describeIf(condition: boolean) {
   return condition ? describe : describe.skip;
 }
 
-export const keys = {
-  backspace: "\uE003"
-};
-
-export async function setup(): Promise<Application> {
-  const app =  new Application({
-    path: AppPaths[process.platform], // path to electron app
-    args: [],
-    startTimeout: 60000,
-    waitTimeout: 10000,
-    env: {
-      CICD: "true"
-    }
+export async function start() {
+  const app = await electron.launch({
+    args: ["--integration-testing"], // this argument turns off the blocking of quit
+    executablePath: AppPaths[process.platform],
+    bypassCSP: true,
   });
 
-  await app.start();
-  // Wait for splash screen to be closed
-  while (await app.client.getWindowCount() > 1);
-  await app.client.windowByIndex(0);
-  await app.client.waitUntilWindowLoaded();
-  await showCatalog(app);
-
-  return app;
-}
+  const window = await app.waitForEvent("window", {
+    predicate: async (page) => page.url().startsWith("http://localhost"),
+  });
 
-export async function showCatalog(app: Application) {
-  await app.client.waitForExist("#hotbarIcon-catalog-entity .Icon");
-  await app.client.click("#hotbarIcon-catalog-entity .Icon");
+  return {
+    app,
+    window,
+    cleanup: async () => {
+      await window.close();
+      await app.close();
+    },
+  };
 }
 
-type AsyncPidGetter = () => Promise<number>;
-
-export async function tearDown(app?: Application) {
-  if (!app?.isRunning()) {
-    return;
-  }
-
-  const pid = await (app.mainProcess.pid as any as AsyncPidGetter)();
-
-  await app.stop();
-
-  try {
-    process.kill(pid, "SIGKILL");
-  } catch (e) {
-    console.error(e);
-  }
+export async function clickWelcomeButton(window: Page) {
+  await window.click("#hotbarIcon-catalog-entity .Icon");
 }
 
-export const promiseExec = util.promisify(exec);
+/**
+ * From the catalog, click the minikube entity and wait for it to connect, returning its frame
+ */
+export async function lauchMinikubeClusterFromCatalog(window: Page): Promise<Frame> {
+  await window.waitForSelector("div.TableCell");
+  await window.click("div.TableCell >> text='minikube'");
+  await window.waitForSelector("div.drawer-title-text >> text='KubernetesCluster: minikube'");
+  await window.click("div.EntityIcon div.HotbarIcon div div.MuiAvatar-root");
 
-type HelmRepository = {
-  name: string;
-  url: string;
-};
+  const minikubeFrame = await window.waitForSelector("#cluster-frame-484e864bad9b84ce5d6b4fff704cc0e4");
```

Comment:
From line 205 in `src/main/catalog-sources/kubeconfig-sync.ts`:

```ts
const clusterId = createHash("md5").update(`${filePath}:${contextName}`).digest("hex");
```

I guess I should probably make it so that the the ID is computed on demand for each OS